src/partition/repart.c

```diff
@@ -6963,6 +6961,40 @@ static int parse_argv(int argc, char *argv[]) {
                         *p = gpt_partition_type_override_architecture(*p, arg_architecture);
         }
 
+        if (arg_signing_provider && !private_key)
+                return log_error_errno(
+                                SYNTHETIC_ERRNO(EINVAL),
+                                "--private-key= must be specified when --signing-provider= is used.");
+
+        if (private_key) {
+                if (isempty(arg_signing_provider)) {
+                        _cleanup_(erase_and_freep) char *k = NULL;
+                        size_t n = 0;
+
+                        r = read_full_file_full(
+                                        AT_FDCWD, private_key, UINT64_MAX, SIZE_MAX,
+                                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                                        NULL,
+                                        &k, &n);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read key file '%s': %m", private_key);
+
+                        EVP_PKEY_free(arg_private_key);
+                        arg_private_key = NULL;
```

Comment:
given this is executed once only now (as this is no longer in the getopt loop), you can drop these two lines